### PR TITLE
ROC-7355 ROC-7761 claim integration tests

### PR DESCRIPTION
**Pull request by andrewwa-kainos**
### JIRA link ###
Parent: https://tools.hmcts.net/jira/browse/ROC-7355
Subtask: https://tools.hmcts.net/jira/browse/ROC-7761

### Change description ###
Add integration tests for the claim component

**Does this PR introduce a breaking change?** (check one with "x")

- [ ] Yes
- [x] No

### CODE REVIEW

File: domain-sample-data/src/main/java/uk/gov/hmcts/cmc/domain/models/sampledata/SampleClaimData.java [hunk 1]
```diff
@@ -22,19 +22,21 @@
...
+    private String externalId = SampleClaim.EXTERNAL_ID;
     private List<Party> claimants;
     private List<TheirDetails> defendants;
     private Payment payment = SamplePayment.builder().build();
     private Amount amount = SampleAmountBreakdown.builder().build();
     private Interest interest = SampleInterest.standard();
     private String reason = "reason";
     private BigInteger feeAmount = BigInteger.valueOf(4000);
-    private String feeAccountNumber = "PBA1234567";
+    private String feeAccountNumber = FEE_ACCOUNT_NUMBER;
```
Comment (daveoh) [inline domain-sample-data/src/main/java/uk/gov/hmcts/cmc/domain/models/sampledata/SampleClaimData.java]: Do we need this?

File: domain-sample-data/src/main/java/uk/gov/hmcts/cmc/domain/models/sampledata/SampleClaimData.java [hunk 2]
```diff
@@ -22,19 +22,21 @@
...
     private StatementOfTruth statementOfTruth;
     private PersonalInjury personalInjury = new PersonalInjury(DamagesExpectation.MORE_THAN_THOUSAND_POUNDS);
-    private String externalReferenceNumber = "CLAIM234324";
+    private String externalReferenceNumber = EXTERNAL_REFERENCE_NUMBER;
```
Comment (daveoh) [inline domain-sample-data/src/main/java/uk/gov/hmcts/cmc/domain/models/sampledata/SampleClaimData.java]: and this?

File: domain-sample-data/src/main/java/uk/gov/hmcts/cmc/domain/models/sampledata/SampleClaimData.java (hunk 1 above)
Comment (andrewwa-kainos) [reply domain-sample-data/src/main/java/uk/gov/hmcts/cmc/domain/models/sampledata/SampleClaimData.java]: Nope. Reverting

File: domain-sample-data/src/main/java/uk/gov/hmcts/cmc/domain/models/sampledata/SampleClaimData.java (hunk 2 above)
Comment (andrewwa-kainos) [reply domain-sample-data/src/main/java/uk/gov/hmcts/cmc/domain/models/sampledata/SampleClaimData.java]: Yes, used in ClaimQueryTest

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/ClaimService.java [hunk 3]
```diff
@@ -141,21 +141,21 @@ public Claim getClaimByExternalId(String externalId, User user) {
             authorisation = user.getAuthorisation();
         }
 
-        return caseRepository.getByClaimReferenceNumber(reference, authorisation);
+        Optional<Claim> byClaimReferenceNumber = caseRepository.getByClaimReferenceNumber(reference, authorisation);
+        return byClaimReferenceNumber;
```
Comment (daveoh) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/services/ClaimService.java]: Redundant local variable

File: domain-sample-data/src/main/java/uk/gov/hmcts/cmc/domain/models/sampledata/SampleClaimData.java [hunk 4]
```diff
@@ -22,8 +22,9 @@
 import static uk.gov.hmcts.cmc.domain.models.sampledata.SampleInterest.noInterestBuilder;
 
 public class SampleClaimData {
+    public static final String EXTERNAL_REFERENCE_NUMBER = "CLAIM234324";
 
-    private UUID externalId = UUID.randomUUID();
+    private String externalId = SampleClaim.EXTERNAL_ID;
```
Comment (justin-r-oliver) [inline domain-sample-data/src/main/java/uk/gov/hmcts/cmc/domain/models/sampledata/SampleClaimData.java]: Why are we using a less strict type?

File: src/integrationTest/java/uk/gov/hmcts/cmc/claimstore/controllers/CalculateInterestAmountTest.java [hunk 5]
```diff
@@ -65,14 +57,11 @@ public void shouldReturnBadRequestWhenAmountIsNegative() throws Exception {
...
             .perform(
-                get(
-                    String.format(
-                        "/interest/calculate?from_date=%s&to_date=%s&rate=%s&amount=%s",
-                        fromDate,
-                        toDate,
-                        rate,
-                        amount
-                    )
+                get("/interest/calculate?from_date={from}&to_date={to}&rate={rate}&amount={amount}",
```
Comment (justin-r-oliver) [inline src/integrationTest/java/uk/gov/hmcts/cmc/claimstore/controllers/CalculateInterestAmountTest.java]: I like this

File: domain-sample-data/src/main/java/uk/gov/hmcts/cmc/domain/models/sampledata/SampleClaimData.java (hunk 4 above)
Comment (andrewwa-kainos) [reply domain-sample-data/src/main/java/uk/gov/hmcts/cmc/domain/models/sampledata/SampleClaimData.java]: I had changed it to make it more manageable, but I ended up removing the changes that consumed it directly and indeed in the `ClaimData` class it's still explicitly a `UUID`.

Reverting

**Pull request merged**